### ISSUE TITLE
Fix request: AvoidInfiniteRecursion - this.field false positive

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#506]
```java
    private Object getFieldValue(int index) {
        final String fieldName = this.getFieldNameAt(index);
        final Object fieldValueObject = this.getFieldValue(fieldName); // <- false positive 
   }
```

**Comment by jborgers** [comment]
It is not the this field that makes the false positive. It is because it the called overloaded method has the same number of arguments as the calling method, and no argument is used in a different position, nor as plural form, nor with calling a method on it, or used as argument in a method.
